lib/pbench/server/auth/__init__.py

```diff
@@ -320,14 +463,31 @@ def token_introspect_offline(
                 "sub": <user_id>
             }
         """
-        return jwt.decode(
-            token, key, algorithms=algorithms, audience=audience, **kwargs
-        )
+        try:
+            return jwt.decode(
+                token,
+                self._pem_public_key,
+                algorithms=[self._TOKEN_ALG],
+                audience=self.client_id,
+                options={
+                    "verify_signature": True,
+                    "verify_aud": True,
+                    "verify_exp": True,
+                },
+            )
+        except (
+            jwt.ExpiredSignatureError,
+            jwt.InvalidSignatureError,
+            jwt.InvalidAudienceError,
+        ) as exc:
+            raise OpenIDTokenInvalid() from exc
 
     def get_userinfo(self, token: str = None) -> JSON:
         """The userinfo endpoint returns standard claims about the authenticated
         user, and is protected by a bearer token.
 
+        FIXME - This method appears unused in the rest of the code.
+
```

Comment:
I don't think we would ever going to call the `userinfo` endpoint as we get pretty much all the user information when we decode the tokens. I added this when I did some experimentation with Keycloak APIs and I thought this might come handy but I realized token decode is enough to get all the user information.

Reply:
Well, on that basis, let's rip out this function, then (which will get rid of the `FIXME` 😉).